### PR TITLE
Add logging for Client's public API calls

### PR DESCRIPTION
**Pull request by vinbarnes**
Addresses #12 

Some examples from running tests:
```json
{"feature": "funnel_test", "response": false, "event": "feature is_enabled", "project_id": "project_id", "klass": "PosthogAPIClient", "level": "info", "timestamp": "2024-06-06T20:37:43.594899Z"}
{"feature": "funnel_test", "response": true, "event": "disable feature", "project_id": "project_id", "klass": "PosthogAPIClient", "level": "info", "timestamp": "2024-06-06T20:37:43.603948Z"}
{"feature": "funnel_test", "response": true, "event": "remove feature", "project_id": "project_id", "klass": "PosthogAPIClient", "level": "info", "timestamp": "2024-06-06T20:36:18.666905Z"}
```

### CODE REVIEW

File: feature_gate/client.py [hunk 1]
```diff
@@ -4,26 +4,35 @@ class FeatureNotFound(ValueError):
...
 
   def remove(self, feature):
+    self.logger.info("remove feature", feature=feature)
     return self.adapter.remove(feature)
 
   def features(self):
     return self.adapter.features()
 
   def is_enabled(self, feature):
+    self.logger.info("feature is_endabled", feature=feature)
```
Comment (thephw) [inline feature_gate/client.py]:
```suggestion
    self.logger.info("feature is_enabled", feature=feature)
```

Need to log the response (true/false) here too

Comment (vinbarnes) [reply feature_gate/client.py]: Typo fixed in [ce9dee5](https://github.com/deftinc/feature_gate/pull/14/commits/ce9dee5ca84ff852b73e958b2a4fec993651ec88).

I was fighting with testing even doing this simplest thing possible. Once some tests were passing in the easy case, the response logging was added in [0d59f04](https://github.com/deftinc/feature_gate/pull/14/commits/0d59f04198ad48d282f819f10b961ca72aef86de).

**Comment by thephw** [comment]
We could probably remove the additional logging in `feature_gate/adapters/memory.py` right? Thats all covered by the new logging?

**Comment by vinbarnes** [comment]
> We could probably remove the additional logging in `feature_gate/adapters/memory.py` right? Thats all covered by the new logging?

I can remove it but I only changed a variable/method name for visibility purposes. It may look like I added those statements in the diff, but they predate the changes in this PR. When [the memory adapter is introduced the logging is already there]( https://github.com/deftinc/feature_gate/blob/23ec10483a755af20796b702d7593b1cbbbf6d89/feature_gate/adapters/memory.py). 

I'm fine removing the logging. Want to be sure it's for the right reason.

**Comment by vinbarnes** [comment]
Logging removed from memory adapter in fb6498e but we can revert if this change is unnecessary. 

See [discussion](https://github.com/deftinc/feature_gate/pull/14#issuecomment-2153824762) ☝️

**Comment by thephw** [comment]
@vinbarnes Yeah I think that logging is just redundant now with these changes was the issue. Lets merge. 👍🏻

**Review by thephw — APPROVED** [review]

**Pull request merged**